replica/database.hh

```diff
@@ -1432,6 +1433,7 @@ private:
 
     friend db::data_listeners;
     std::unique_ptr<db::data_listeners> _data_listeners;
+    std::unique_ptr<db::toppartitions_data_listener> _tp_listener;
```

Comment:
Why not use the member already available above?

Reply:
Because the member above of the type `data_listeners` is an abstraction of a `set` of a generic `data_listener` and what I need here is a `toppartitions_data_listener` in order to persistently sample the toppartitions.